From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rachapalli Bandi Jagadeesh <jagadeesh.rachapalli.bandi@intel.com>
Date: Tue, 14 Oct 2025 15:31:31 +0530
Subject: [PATCH]  Handle __dma_fence_is_later() prototype change
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

__dma_fence_is_later() args is taking struct dma_fence instead of ops
from 6.17 kernel.
Hence observing below warnings.
To resolve these warnings, started using ops pointer instead of
dma_fence for below 6.17 kernels.

Warnings:
-----------------------------------------------------------------------
warning: passing argument 1 of ‘__dma_fence_is_later’ makes integer
from pointer without a cast [-Wint-conversion]
warning: passing argument 3 of ‘__dma_fence_is_later’ makes pointer
from integer without a cast [-Wint-conversion]
-----------------------------------------------------------------------

Reference:
        549810e91815 dma-fence: Change signature of __dma_fence_is_later

Signed-off-by: Rachapalli Bandi Jagadeesh <jagadeesh.rachapalli.bandi@intel.com>
---
 drivers/gpu/drm/xe/xe_hw_fence.c  |  5 +++++
 drivers/gpu/drm/xe/xe_sched_job.c | 12 ++++++++++++
 2 files changed, 17 insertions(+)

diff --git a/drivers/gpu/drm/xe/xe_hw_fence.c b/drivers/gpu/drm/xe/xe_hw_fence.c
index b2a0c46..0d20bce 100644
--- a/drivers/gpu/drm/xe/xe_hw_fence.c
+++ b/drivers/gpu/drm/xe/xe_hw_fence.c
@@ -167,8 +167,13 @@ static bool xe_hw_fence_signaled(struct dma_fence *dma_fence)
 	struct xe_device *xe = fence->xe;
 	u32 seqno = xe_map_rd(xe, &fence->seqno_map, 0, u32);
 
+#ifdef BPM_DRM_FENCE_IS_LATER_ARG_DMA_FENCE_OPS_NOT_PRESENT
+	return dma_fence->error ||
+                !__dma_fence_is_later(dma_fence->seqno, seqno, dma_fence->ops);
+#else
 	return dma_fence->error ||
 		!__dma_fence_is_later(dma_fence, dma_fence->seqno, seqno);
+#endif
 }
 
 static bool xe_hw_fence_enable_signaling(struct dma_fence *dma_fence)
diff --git a/drivers/gpu/drm/xe/xe_sched_job.c b/drivers/gpu/drm/xe/xe_sched_job.c
index d21bf8f..ea1a113 100644
--- a/drivers/gpu/drm/xe/xe_sched_job.c
+++ b/drivers/gpu/drm/xe/xe_sched_job.c
@@ -220,9 +220,15 @@ bool xe_sched_job_started(struct xe_sched_job *job)
 	struct dma_fence *fence = dma_fence_chain_contained(job->fence);
 	struct xe_lrc *lrc = job->q->lrc[0];
 
+#ifdef BPM_DRM_FENCE_IS_LATER_ARG_DMA_FENCE_OPS_NOT_PRESENT
+        return !__dma_fence_is_later(xe_sched_job_lrc_seqno(job),
+                                     xe_lrc_start_seqno(lrc),
+                                     fence->ops);
+#else
 	return !__dma_fence_is_later(fence,
 				     xe_sched_job_lrc_seqno(job),
 				     xe_lrc_start_seqno(lrc));
+#endif
 }
 
 bool xe_sched_job_completed(struct xe_sched_job *job)
@@ -235,9 +241,15 @@ bool xe_sched_job_completed(struct xe_sched_job *job)
 	 * parallel handshake is done.
 	 */
 
+#ifdef BPM_DRM_FENCE_IS_LATER_ARG_DMA_FENCE_OPS_NOT_PRESENT
+        return !__dma_fence_is_later(xe_sched_job_lrc_seqno(job),
+                                     xe_lrc_seqno(lrc),
+                                     fence->ops);
+#else
 	return !__dma_fence_is_later(fence,
 				     xe_sched_job_lrc_seqno(job),
 				     xe_lrc_seqno(lrc));
+#endif
 }
 
 void xe_sched_job_arm(struct xe_sched_job *job)
-- 
2.43.0

